lib/galaxy/tools/parameters/validation.py

```diff
@@ -391,35 +299,19 @@ class MetadataInFileColumnValidator(Validator):
 
     requires_dataset_metadata = True
 
-    @classmethod
-    def from_element(cls, param, elem):
-        filename = elem.get("filename")
-        assert filename, f"Required 'filename' attribute missing from {elem.get('type')} validator."
-        filename = f"{param.tool.app.config.tool_data_path}/{filename.strip()}"
-        assert os.path.exists(filename), f"File {filename} specified by the 'filename' attribute not found"
-        metadata_name = elem.get("metadata_name")
-        assert metadata_name, f"Required 'metadata_name' attribute missing from {elem.get('type')} validator."
-        metadata_name = metadata_name.strip()
-        metadata_column = int(elem.get("metadata_column", 0))
-        split = elem.get("split", "\t")
-        message = elem.get("message", f"Value for metadata {metadata_name} was not found in {filename}.")
-        line_startswith = elem.get("line_startswith")
-        if line_startswith:
-            line_startswith = line_startswith.strip()
-        negate = elem.get("negate", "false")
-        return cls(filename, metadata_name, metadata_column, message, line_startswith, split, negate)
-
     def __init__(
         self,
-        filename,
-        metadata_name,
-        metadata_column,
-        message="Value for metadata not found.",
-        line_startswith=None,
-        split="\t",
-        negate="false",
+        filename: str,
+        metadata_name: str,
+        metadata_column: int,
+        message: str,
+        line_startswith: Optional[str] = None,
+        split: str = "\t",
+        negate: bool = False,
     ):
         super().__init__(message, negate)
+        assert filename
+        assert os.path.exists(filename), f"File {filename} specified by the 'filename' attribute not found"
```

Comment:
Is that really needed here as assert?

Reply:
It was present in the existing validation code. I guess the idea is to not load the tool if that file is not available.